### PR TITLE
chore: Use python with dbus in pidgin

### PR DESCRIPTION
**Pull request by pniedzwiedzinski**
###### Motivation for this change
Fix `purple-*` python scripts, which require dbus package.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117408` at b4946f56 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>25 packages built successfully:</summary>
  <ul>
    <li>pidgin</li>
    <li>pidgin-carbons</li>
    <li>pidgin-latex</li>
    <li>pidgin-mra</li>
    <li>pidgin-msn-pecan</li>
    <li>pidgin-opensteamworks</li>
    <li>pidgin-osd</li>
    <li>pidgin-otr</li>
    <li>pidgin-sipe</li>
    <li>pidgin-skypeweb</li>
    <li>pidgin-window-merge</li>
    <li>pidgin-xmpp-receipts</li>
    <li>purple-discord</li>
    <li>purple-facebook</li>
    <li>purple-hangouts</li>
    <li>purple-lurch</li>
    <li>purple-matrix</li>
    <li>purple-plugin-pack</li>
    <li>purple-slack</li>
    <li>purple-vk-plugin</li>
    <li>purple-xmpp-http-upload</li>
    <li>skype4pidgin</li>
    <li>telegram-purple</li>
    <li>telepathy-haze</li>
    <li>toxprpl</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

intltool is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/networking/instant-messengers/pidgin/default.nix:32:3:
```
   |
32 |   buildInputs = let
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/instant-messengers/pidgin/default.nix:54:15:
```
   |
54 |   patches = [ ./pidgin-makefile.patch ./add-search-path.patch ];
   |               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/instant-messengers/pidgin/default.nix:54:39:
```
   |
54 |   patches = [ ./pidgin-makefile.patch ./add-search-path.patch ];
   |                                       ^
```
</li>
  </ul>
</details>

**Review by vcunat — APPROVED** [review]

File: pkgs/applications/networking/instant-messengers/pidgin/default.nix [hunk 1]
```diff
@@ -29,11 +29,13 @@ let unwrapped = stdenv.mkDerivation rec {
...
 
-  buildInputs = [
+  buildInputs = let
+    python-with-dbus = python.withPackages (pp: with pp; [ dbus-python ]);
+  in [
     aspell startupnotification
     gst_all_1.gstreamer gst_all_1.gst-plugins-base gst_all_1.gst-plugins-good
     libxml2 nss nspr
-    libXScrnSaver ncurses python
+    libXScrnSaver ncurses python-with-dbus
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/pidgin/default.nix:38]:
```suggestion
    libXScrnSaver ncurses python
```

File: pkgs/applications/networking/instant-messengers/pidgin/default.nix [hunk 2]
```diff
@@ -29,11 +29,13 @@ let unwrapped = stdenv.mkDerivation rec {
 
   NIX_CFLAGS_COMPILE = "-I${gst_all_1.gst-plugins-base.dev}/include/gstreamer-1.0";
 
-  buildInputs = [
+  buildInputs = let
+    python-with-dbus = python.withPackages (pp: with pp; [ dbus-python ]);
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/pidgin/default.nix:33]:
```suggestion
    python = python.withPackages (pp: with pp; [ dbus-python ]);
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/pidgin/default.nix (hunk 2 above)
Comment (vcunat) [reply pkgs/applications/networking/instant-messengers/pidgin/default.nix:33]: It's used just once, so we wouldn't even need to name it, i.e. we could just put it there directly without a let-binding.

Comment (SuperSandro2000) [reply pkgs/applications/networking/instant-messengers/pidgin/default.nix:33]: It is probably cleaner when we name it but I don't see a reason to give it a special name.

Comment (vcunat) [reply pkgs/applications/networking/instant-messengers/pidgin/default.nix:33]: Ah, but your suggestion won't work due to infinite recursion (`let` is always recursive).

**Pull request merged**